Makefile

```diff
@@ -286,15 +304,13 @@ $(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATF
 assets/osx/installer.tiff: assets/osx/installer.png assets/osx/installer@2x.png
 	tiffutil -cathidpicheck $^ -out $@
 
-$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-$(APPLICATION_VERSION)-darwin-$(TARGET_ARCH).dmg: assets/osx/installer.tiff \
-	| $(BUILD_DIRECTORY)
+electron-installer-dmg: assets/osx/installer.tiff
```

Comment:
Yeah, the main motivation behind this change is that with electron-builder, is the tool what determines the final file name. Trying to keep the electron-builder configuration in sync with the target name involved a lot of work (e.g. to match how the injected extra metadata affects the final package, etc), and we don't need to reference this files anywhere else.

What would be the benefits of using the actual paths as Makefile targets? Should we go for it once we do have a Makefile rule that needs to reference it?